File: problems.js
//problem solving

console.log(`Saqlain Aslam`);

// Problem 1
// Interchange the value of first index and second last index of an array?
let arr1 = [2, 3, 4, 56, 6];
console.log(arr1);

// solution 1 using third variable
console.log(`Third variable`);

let temp = arr1[0];
arr1[0] = arr1[4];
arr1[4] = temp;
console.log(`${arr1}`);

//solution2
console.log(`Array Destructuring`);

[arr1[0], arr1[4]] = [arr1[4], arr1[0]];
console.log(arr1);

//solution3
console.log(`Array Splice method`);
// Splice method is used to modify the array by addding, remmoving and replacing the content of array
arr1.splice(0, 1, arr1[4]);
arr1.splice(4, 1, 2);
console.log(arr1);

// Problem 2
// Make the length of array dynamic and take the inputs and length of array from the user

//solution
// let length = parseInt(prompt("Enter the length of the array:"));
// let arr2 = [];

// for (let index = 0; index < length; index++) {
//   let input = prompt(`Enter elemen ${index + 1}:`);
//   arr2.push(input);
// }

// console.log(`Dynamic Array is: ${arr2}`);

// Problem 3
// Replace the two index of an array and the index position will be selected by the user.

// let arr3 = [2, 5, 7, 8, 9, 5];

// let index1 = parseInt(prompt(`Enter first element:`));
// let index2 = parseInt(prompt(`Enter second element:`));

// if (
//   isNaN(index1) ||
//   isNaN(index2) ||
//   index1 < 0 ||
//   index2 < 0 ||
//   index1 > arr3.length ||
//   index2 > arr3.length
// ) {
//   console.log(`Your index does not exist`);
// } else {
//   let temp1 = arr3[index1];
//   arr3[index1] = arr3[index2];
//   arr3[index2] = temp1;

//   console.log(`New array after swap is: ${arr3}`);
// }

// Problem 4

File: problems.py
 # 1  swap first element with second last element of an array

arr1 = [10, 20, 30, 40, 50]

temp = arr1[0]
arr1[0] = arr1[-2]  
arr1[-2] = temp

print("after swap:", arr1)


# 2 dynamic array 

length = int(input("enter length of array: "))
arr2 = []

for i in range(length):
    num = int(input(f"Enter element {i+1}: "))
    arr2.append(num)

if length >= 2: 
    temp = arr2[0]
    arr2[0] = arr2[-2]
    arr2[-2] = temp

print("after swap first and second last:", arr2)



# 3 swapuser-selected indexes in array

arr3 = [5, 10, 15, 20, 25]

print("original aray:", arr3)
i = int(input("Enter first index to swap: "))
j = int(input("Enter second index to swap: "))

# check if indexes are valid
if i <= 0 < len(arr3) and j <= 0 < len(arr3):
    temp = arr3[i]
    arr3[i] = arr3[j]
    arr3[j] = temp
    print("array after swapping:", arr3)
else:
    print("invalid indexes")


# 4: count vowels and consonants

text = input("enter string: ").lower()

vowels = "aeiou"
vowel_count = 0
consonant_count = 0

for ch in text:
    if ch.isalpha():  # check if character is a letter
        if ch in vowels:
            vowel_count += 1
        else:
            consonant_count += 1

print("total vowels:", vowel_count)
print("total consonants:", consonant_count)


# 5 sum of matching elements in an array

arr = [1, 2, 3, 2, 4, 1, 2]
num = int(input("Enter number to find sum of all matches: "))

total = 0
for x in arr:
    if x == num:
        total += x

print(f"total of all {num}'s in array:", total)


